### ISSUE TITLE
non-fixed array dimension in `Buildings.HeatTransfer.Data.GlazingSystems.Generic`

### ISSUE THREAD

**Issue by yfuerst** [lbl-srg/modelica-buildings#1134]
I developed a complex building model based on `Buildings.ThermalZones.Detailed.MixedAir`, which simplifies the whole process of parameterization significantly. The single rooms are connected automatically according to the blueprint of the building to be modeled using plain Modelica code. This is achieved using an incidence matrix to define adjacent rooms.

Unfortunately, I am currently forced to use a manipulated version of the Buildings library due to the implementation of `Buildings.HeatTransfer.Data.GlazingSystems.Generic`. The parameters `glass[:]` and `gas[:]` are defined as non-fixed arrays. The array dimensions result from the final definition e.g.
```
record SingleClear3 = Buildings.HeatTransfer.Data.GlazingSystems.Generic (
       glass={Glasses.ID102()},
       UFra=3) "Single pane, clear glass 3mm"
```
Apparently, Modelica doesn't like this kind of implementation, when the parameter `Buildings.HeatTransfer.Data.GlazingSystems.Generic` is manipulated between several layers of inheritance. I was forced to add additional parameters for the array dimensions (`nGla` and `nGas` for `glass[nGla]` and `gas[nGas]` respectively) like it is already done in `Buildings.HeatTransfer.Data.OpaqueConstructions.Generic` to get it working.

Is it possible to adapt the implementation of `Buildings.HeatTransfer.Data.GlazingSystems.Generic` with `Buildings.HeatTransfer.Data.OpaqueConstructions.Generic` in future versions of Modelica Buildings?

**Comment by mwetter** [comment]
@yfuerst What change of the code would you need, and does this still allow windows with any number of panes (e.g., the number of panes is deduced from the record.

Also, is this a bug in the library or a limitation of a Modelica tool (if the latter, which tool)? How many layers of inheritance should not affect whether the model can be translated.

**Comment by yfuerst** [comment]
Despite being a big fan of JModelica, I developed and tested the model in Dymola. Unfortunately, I am not very familiar with the limitations between the different Modelica tools in detail.

Now comes the embarrassing part:
The model I developed depends on Buildings 4.0 (I could have mentioned this in the first place). I was just working on a small working example to illustrate my problem, when I realized it disappeared once I switched to Buildings 5.0.2 (the current development branch). So I guess this can be closed.
